### PR TITLE
Update Developer guide

### PR DESCRIPTION
**Pull request by froststein**
Integrate feedback from peer review.

Add explanation for SessionEditCommand.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -9,29 +12,47 @@
...
 * [User Stories](#user-stories)
 * [Non-Functional Requirements](#non-functional-requirements)
 * [Glossary](#glossary)
 
-
 ## Acknowledgements
 * Address Book (Level-3) - Provide samples of User Guide(UG) and Developer Guide(DG)
 
+## Introduction
+The guide's purpose is to allow readers to understand how the application works internally, so that future engineers could continue working on the project with ease.
```
Comment (matheril) [inline docs/DeveloperGuide.md:41]: Consider changing "could" to "can"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -9,29 +12,47 @@
...
 * [Glossary](#glossary)
 
-
 ## Acknowledgements
 * Address Book (Level-3) - Provide samples of User Guide(UG) and Developer Guide(DG)
 
+## Introduction
+The guide's purpose is to allow readers to understand how the application works internally, so that future engineers could continue working on the project with ease. 
+
+### How to use this developer guide?
```
Comment (matheril) [inline docs/DeveloperGuide.md:43]: The question mark at the end is unnecessary. Consider removing it.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -9,29 +12,47 @@
...
+[Profile Component](#profile-component), [TextUI Component](#textui-component),
+[Storage Component](#storage-component), [Parser Component](#parser-component) and [Command Component](#command-component)
+* Each component has its own section in this guide explaining them in greater detail.
+* There is also a [diagram](#interaction-between-components) that shows the inner workings of how each component interacts with each other inside SplitLah.
+* SplitLah supports a total of **18 commands**. However, in this developer guide, we would not explain in detail how the `help` and `exit` command works. 
+To view the usages of SplitLah's commands please refer to our [userguide](https://ay2122s2-cs2113t-t10-1.github.io/tp/UserGuide.html).
+* The explanation of each command starts [here](#implementation).
+* In each command explanation section, there is an API reference link that brings the reader to the source code of the command. 
+* API reference refers to the main source code that is responsible for carrying out the command's action.
+
```
Comment (matheril) [inline docs/DeveloperGuide.md:55]: Consider rewording this section to:
SplitLah is divided into eight main components: Manager, Profile, TextUI... (consider making each one a subpoint with deeper indentation)
Each component has its own section in this guide explaining them in detail.
This [diagram] shows the inner workings of how each component interacts with each other.
Splitlah supports a total of 18 commands. However, the `help` and `exit` commands will not be explained in detail. (Consider if saying "we" is necessary here; not sure. It's also fine to remove the sentence about help and exit as this sentence isn't really that important for the reader to know).
Please refer to the [userguide] to find out how to use each command.
Please refer to the [implementation section] to find out how each command is designed and implemented in SplitLah.
Each command's section in this guide contains an API reference link that brings you to the main source code responsible for executing the command.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -99,6 +119,21 @@ As TextUI handles all input and output streams, these streams can be changed wit
 ### Storage Component
 ![Storage Component Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/StorageComponent.drawio.png)
 <br>
+The `Storage` class is initialized by the `Manager` class when the application starts. 
+Upon initialization, the `Storage` class will check if there were pre-existing data stored by the user. 
+* If a data file is found, the `Storage` class would retrieve the data and save it 
+to the `Profile` object that is managed by the `Manager` class in the current run of SplitLah. 
+* Else a new data file is created and an empty `Profile` instance is created instead.
+
```
Comment (matheril) [inline docs/DeveloperGuide.md:127]: Consider rewording this to (formatting is scuffed here, but should give a rough idea of the indentation)
The `Storage` class is initialized (Might want to standardize UK/US spelling) by the `Manager` class when the application starts.
Upon initialization, the `Storage` class checks if a save file already exists. A save file records user data related to SplitLah even after the application is closed.
 * If a save file is found, the `Storage` class loads data from it into the `Profile` object managed by the `Manager` object (I believe Manager is instantiated so object might be better to say than class. The first point already says "when the application starts" so I don't think it's necessary to emphasize that SplitLah is running).
 * Else, a new save file is created and an empty `Profile` object is created instead. (standardise object/instance)

Only commands that update attributes of the `Profile` object also ensure that the save file is updated to record these changes. They are:
*  create
*  delete
*  edit

Once the command finishes executing, `Manager#saveProfile` is called to update the save file.

*  For example: 
   *      The user enters `group /create /n Highschool Friends /pl Alice Bob`. 
   *      SplitLah recognises this as a `GroupCreateCommand` and creates a new group accordingly.
   *      As this updates the list of groups stored in the `Profile` object, `Manager#saveProfile` is called to update the save file with the newly created group.

All classes associated (not sure if there's a better word to replace "under" that more clearly describes the relationship between these classes?) the `Profile` component implement the `Serializable` class. This allows the `Storage` class to write into the save file without having to pre-process the data.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -254,7 +289,39 @@ session unique identifier from the list of sessions.
...
+3. The `SessionEditCommand#run` method is then invoked to run the `session /edit` command.
+4. The list of sessions are stored in a `Profile` object, hence `Manager#getProfile` is called
+   before the list of sessions can be retrieved.
+5. Once the `Profile` object is returned, `Profile#getSession` is called to retrieve the `Session` object with the specified
+   session unique identifier from the list of sessions.
+   * If a `Session` object with the specified session unique identifier cannot be found, it prints the error message and returns control to `SplitLah`.
+   * Else the `Session` object with the specified session unique identifier is returned.
+6. The detail of how a session is updated in the reference diagram below.
+![Reference Frame Update Session Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/RefUpdateSession.png)
+7. `SessionEditCommand#run` would check if there is an update for a new list of persons, new session or new session date.
```
Comment (matheril) [inline docs/DeveloperGuide.md:316]: Try to reword this section to present tense

**Review by matheril — CHANGES_REQUESTED** [review]
Content looks good but changes to phrasing and tense recommended.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -254,7 +299,39 @@ session unique identifier from the list of sessions.
...
+6. The detail of how a session is updated in the reference diagram below.
+![Reference Frame Update Session Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/RefUpdateSession.png)
+7. `SessionEditCommand#run` will check if there is an update for a new list of persons, new session or new session date.
+   * If there is an update on the list of persons. It would first check if the newly provided list of persons contains duplicated names.
+     * If duplicated names are detected, an error message would be printed and control is returned to `SplitLah`.
+     * Else, it would call `PersonList#isSuperSet` to check if the newly supplied list of persons contain all existing persons in the session.
+     * If `PersonList#isSuperSet` returns `true`, it would then call `Session#addPerson` to add in the new list of persons.
+   * If there is an update on the session name. It would call `Session#setSessionName` to set the new session name.
+   * If there is an update on the session date. It would call `Session#setDateCreated` to set the new session date.
```
Comment (warrencxw) [inline docs/DeveloperGuide.md:332]: Consider rephrasing this to 
If ...,  `Session#someMethod` is called to set ...

**Review by warrencxw — CHANGES_REQUESTED** [review]
Added a few suggestions to improve the readability.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -9,29 +12,49 @@
...
+
+### How to use this developer guide
+* SplitLah is divided into **six components**: [Manager Component](#manager-component)
+[Profile Component](#profile-component), [TextUI Component](#textui-component),
+[Storage Component](#storage-component), [Parser Component](#parser-component) and [Command Component](#command-component)
+* Each component has its own section in this guide explaining them in detail.
+* This [diagram](#interaction-between-components) shows the inner workings of how each component interacts with each other.
+* SplitLah supports a total of **18 commands**. However, the `help` and `exit` commands will not be explained in detail.
+* Please refer to the [userguide](https://ay2122s2-cs2113t-t10-1.github.io/tp/UserGuide.html) to find out how to use each command.
+* Please refer to [here](#implementation) to find out how each command is designed and implemented in SplitLah.
```
Comment (ivanaitzliddat) [inline docs/DeveloperGuide.md:55]: Perhaps can change "here" to "implementation section" or maybe "please refer here"

**Review by ivanaitzliddat — APPROVED** [review]
Looks good to merge but left a comment for better phrasing

**Review by matheril — APPROVED** [review]
Looks good to merge

**Pull request merged**